scripts/helpers/read-codex-config.py

```
@@ -0,0 +1,204 @@
+    seen = set()
+    for raw_line in path.read_text(encoding="utf-8").splitlines():
+        line = _strip_comment(raw_line)
+        if not line:
+            continue
+        if line.startswith("["):
+            if not line.endswith("]") or line.startswith("[["):
+                raise ConfigError("unsupported TOML table")
+            section = _key_path(line[1:-1])
+            continue
+        if "=" not in line:
+            raise ConfigError("invalid TOML assignment")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)